lib/ansible/modules/cloud/azure/azure_rm_webapp_facts.py

```diff
@@ -0,0 +1,292 @@
+        except CloudError as exc:
+            self.fail("Error listing web apps in resource groups {0} - {1}".format(self.resource_group, str(exc)))
+
+        results = []
+        for item in response:
+            if self.has_tags(item.tags, self.tags):
+                pip = self.serialize_obj(item, AZURE_OBJECT_CLASS)
+                site_config = self.list_webapp_configuration(self.resource_group, item.name)
+                app_settings =  self.list_webapp_appsettings(self.resource_group, item.name)
+                curated_output = self.construct_curated_webapp(pip, site_config, app_settings)
+                results.append(curated_output)
+        return results
+
+    def list_all(self):
+        self.log('List web apps in current subscription')
+        try:
+            response = list(self.web_client.web_apps.list())
+        except CloudError as exc:
+            self.fail("Error listing web apps: {1}".format(str(exc)))
+
+        results = []
+        for item in response:
+            if self.has_tags(item.tags, self.tags):
+                pip = self.serialize_obj(item, AZURE_OBJECT_CLASS)
+                site_config = self.list_webapp_configuration(pip['properties']['resourceGroup'], item.name)
+                app_settings =  self.list_webapp_appsettings(pip['properties']['resourceGroup'], item.name)
+                curated_output = self.construct_curated_webapp(pip, site_config, app_settings)
+                results.append(curated_output)
+        return results
+
+    def list_webapp_configuration(self, resource_group, name):
+        self.log('Get web app {0} configuration'.format(name))
+
+        response = []
+
+        try:
+            response = self.web_client.web_apps.get_configuration(resource_group_name=resource_group, name=name)
+        except CloudError as ex:
+            self.fail('Error getting web app {0} configuration'.format(name))
+
+        return response.as_dict()
+
+    def list_webapp_appsettings(self, resource_group, name):
+        self.log('Get web app {0} app settings'.format(name))
+
+        response = []
+
+        try:
+            response = self.web_client.web_apps.list_application_settings(resource_group_name=resource_group, name=name)
+        except CloudError as ex:
+            self.fail('Error getting web app {0} app settings'.format(name))
+
+        return response.as_dict()
+
+    def construct_curated_webapp(self, webapp, configuration=None, app_settings=None, deployment_slot=None):
+        curated_output = dict()
+        curated_output['id'] = webapp['id']
+        curated_output['name'] = webapp['name']
+        curated_output['resource_group'] = webapp['properties']['resourceGroup']
+        curated_output['location'] = webapp['location']
+        curated_output['plan'] = webapp['properties']['serverFarmId']
+        curated_output['tags'] = webapp.get('tags', None)
+
+        # add properties
+        curated_output['properties'] = webapp['properties']
```

Comment:
what is properties exactly? shouldn't be flattened?

Reply:
as offline disucssed with you, some important properties not in module input, but model output, such as hostname, ip etc. why flatten it?  it's more clean to wrap it properties based 2 reasons: 1. seperated with curated output. 2.  consistent with model  output

Reply:
just don't know exactly what's in the properties :-)

Reply:
updated documentation